custom_components/tuya_local/devices/nac_ap470_purifier.yaml

```diff
@@ -0,0 +1,146 @@
+name: Air purifier
+products:
+  - id: y7iiubdc5sufjvo3
+    name: "NAC AP470 Air Purifier"
+primary_entity:
+  entity: fan
+  translation_key: fan_with_presets
+  icon: "mdi:air-purifier"
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 4
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: 1
+          value: Auto
+        - dps_val: 2
+          value: Sleep
+        - dps_val: 3
+          value: Level 1
+        - dps_val: 4
+          value: Level 2
+        - dps_val: 4
+          value: Level 3
+        - dps_val: 6
+          value: Level 4
+    - id: 21
+      name: fault_code
+      type: bitfield
+secondary_entities:
+  - entity: sensor
+    class: pm25
+    dps:
+      - id: 2
+        type: integer
+        name: sensor
+        class: measurement
+        unit: ugm3
+  - entity: sensor
+    name: Filter left days
+    icon: "mdi:air-filter"
+    category: diagnostic
+    dps:
+      - id: 16
+        type: integer
+        name: sensor
+        unit: "d"
+  - entity: sensor
+    name: Total time
+    icon: "mdi:air-filter"
+    category: diagnostic
+    dps:
+      - id: 17
+        type: integer
+        name: sensor
+        unit: "h"
+  - entity: switch
+    name: Ionizer
+    icon: "mdi:atom-variant"
+    category: config
+    dps:
+      - id: 6
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+          - dps_val: false
+  - entity: lock
+    name: Child lock
+    category: config
+    icon: "mdi:hand-back-right-off"
+    dps:
+      - id: 7
+        type: boolean
+        name: lock
+  - entity: switch
+    name: Light
+    category: config
+    icon: "mdi:lightbulb"
+    dps:
+      - id: 8
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+          - dps_val: false
+  - entity: switch
+    name: UV lamp
+    category: config
+    icon: "mdi:shield-bug"
+    dps:
+      - id: 9
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+          - dps_val: false
+  - entity: switch
```

Comment:
Is this a switch, or a binary_sensor (an indicator to change/clean the filter)

Reply:
In tuya app this is switch. On some interval this switch automaticaly turns on and there is an icon on display. Tu turn it off you have to switch off this in app.